### ISSUE TITLE
Supporting new Backend Dependencies fields

### ISSUE THREAD

**Issue by AlexanderWert** [elastic/apm-server#7656]
Meta Issue for Backend Dependencies requirements

* [x] https://github.com/elastic/apm-server/issues/7715
* [x] https://github.com/elastic/apm-server/issues/7734
* [x] https://github.com/elastic/apm-server/issues/7745

**Comment by simitt** [comment]
@lahsivjar @SylvainJuge could you summarize what and how to still manually test for this feature and what has been tested already.

**Comment by SylvainJuge** [comment]
I did all of my tests with opbeans-java + java agent + trigger a transaction on `http://localhost:8080/api/orders` that calls the database in the following configurations:
1. current released agent `1.31.0` + new apm-server built from `main` running locally
2. WIP agent with new fields PR https://github.com/elastic/apm-agent-java/pull/2578 + new apm-server built from `main` running locally
3. WIP agent with new fields + current APM server (8.2.0 on cloud)

For 1) apm-server should infer the `service.target.type` = `h2` and `service.target.name` = absent, in UI the map and dependencies should still appear with `h2` name as it's the value used for `.resource` field.

For 2) apm-server should store the provided `service.target.type` = `h2` and `service.target.name` = `OPBEANS` that is provided by the agent, the dependencies and map still show `h2/OPBEANS`.

For 3) apm-server should just ignore the new fields and the map and dependencies should show `h2/OPBEANS` so everything works for older apm-servers.

I haven't checked in-depth if the dependencies metrics have all the new fields, but this should be quite easy to check in discover. Let me know if you need more details or help to further test this.

**Comment by simitt** [comment]
Whoever picks this up, for further testing refer to @lahsivjar 's descriptions on
* https://github.com/elastic/apm-server/pull/8155
* https://github.com/elastic/apm-server/pull/7925
* https://github.com/elastic/apm-server/pull/7924
* https://github.com/elastic/apm-server/pull/7870

**Comment by marclop** [comment]
https://github.com/elastic/apm-server/pull/7925, https://github.com/elastic/apm-server/pull/7924 and https://github.com/elastic/apm-server/pull/7870 (Requisite of by 7924) were tested and verified by @stuartnelson3, the only remaining to test is:

- [ ] https://github.com/elastic/apm-server/pull/8155

**Comment by lahsivjar** [comment]
#8155 should be tested by testcase 3.2 of #7925

**Comment by axw** [comment]
Given the above, I'll mark this as done.
